# Pokemón

## Tarea 3:
## Juego de cartas pokemón, combinaciones de mano

In [55]:
library(prob)

Pokémon: Juego de Cartas Coleccionables es un juego de cartas coleccionables competitivo para dos jugadores basado en la saga de videojuegos Pokémon.


En una partida de Pokémon Trading Card Game, los jugadores asumen el papel de un entrenador Pokémon durante un combate de forma análoga a los videojuegos. Para poder jugar, cada jugador debe tener una baraja de 60 cartas que habrá preparado previamente. La composición de las barajas es libre.

**Fuente:**

[Pokémon Trading Card Game](https://es.wikipedia.org/wiki/Pokémon_Trading_Card_Game)

Tomando en cuenta que existen 801 pokemones en nuestra base de datos ¿Cuantas posibles manos existen dentro del juego Pokemón?

In [19]:
#se escogen 60 cartas de un total de 801
# se trata de una combinación 801 C  60

choose(801, 60)

[1] 2.057847e+91

In [20]:
cat('Existe un total de:', choose(801, 60) , 'distintas manos dentro del juego')

Existe un total de: 2.057847e+91 distintas manos dentro del juego

**Existen practicamente una infinidad de posibles manos al jugar este juego.**

## Inicio de Juego

"En primer lugar, los jugadores deberán barajar sus respectivas barajas. A continuación, cada uno robará 7 cartas que se quedará en su mano sin mostrarlas al contrincante. Si alguno de los dos jugadores no posee en su mano al menos un Pokémon básico (es decir, un Pokémon que no sea evolución de otro), entonces deberá mostrar sus cartas al oponente para demostrarlo; devolver sus cartas, barajar de nuevo, y robar 7 cartas. El oponente, a cambio, puede robar una carta adicional. El proceso se debe de hacer hasta que ambos jugadores posean al menos en su mano un Pokémon básico." (wikipedia)

In [2]:
#se escogen 7 cartas de un total de 60
# se trata de una combinación 60 C  7
choose(60, 7)

[1] 386206920

**Con las 60 cartas de incio se pueden tener 386,206,920 formas distintas de iniciar una partida**

Si la regla de inicio dice que debes de tener al menos un Pokémon básico, ¿cuál es la probabilidad de escogerlo entre las 7 primeras cartas de inicio si solo tienes uno de ellos en tus 60 cartas?

**devido a que sacar 7 cartas en sucesión y sin reemplazo no son eventos independientes la probabilidad de sacar una determinada carta se puede calcular con la siguente expresion**

$$ P(S) = P(\{e_1, e_2, \dots , e_n\}) = P(e_1 \cup e_2 \cup \dots \cup e_n) = \displaystyle \sum_{i=1}^n P(e_i)$$

**pero se debe de tomar en cuena que se tratan de una Probabilidad condicional para cada evento**


**si solo Sí solo se tiene una carta de Pokémon básico, la probabilidad de sacarla en un mazo de 60 cartas, eligiendo 7 al azar es:**

In [47]:
7/60

[1] 0.1166667

**Dado que hay una penalización si no se obtiene una carta de Pokémon básico, la probabilidad obtenida 0.12 al tener solo una en el mazo es muy baja**

con el teorema de bayes podemos responder a preguntas como estas:

si un jugador que cuenta con 3 cartas en su mazo, ya eligio 4 cartas y no tiene todavia una carta de Pokémon básico, cual es la probabilidad de que pueda sacarla en las tres siguientes cartas, 

**Calculo de sacar una carta de Pokémon básico en las 56 cartas restantes considerando que hay 3 Pokémon básico en el mazo**

In [ ]:
**probabilidad que salga en la primera carta**

In [49]:
3/56

[1] 0.05357143

**probabilidad que salga en la segunda carta**
$P(E2 \cap E1^c) = P(E2|E1^c)P(E1^c) $


In [51]:
(3/55)*(1-3/56)

[1] 0.05162338

**probabilidad que salga en la tercera carta**
$P(E3 \cap E1^c \cap E2^c) = P(E3|E1^c \cap E2^c)P(E1^c \cap E2^c) = P(E3|E1^c \cap E2^c) P(E2^c|E1^c)P(E1^c) $

In [52]:
(3/54)*(1-(3/55)*(1-3/56))*(1-3/56)

[1] 0.04986504

**la probibilidad resultante es**

In [53]:
3/56 + (3/55)*(1-3/56) + (3/54)*(1-(3/55)*(1-3/56))*(1-3/56)

[1] 0.1550598